File: ruby/lesson 1/binary_search.rb
def binary_search(nums, target)
    lo = 0
    hi = nums.length - 1

    while lo <= hi
        mid = lo + (hi - lo) / 2
        next hi = mid - 1 if nums[mid] > target
        next lo = mid + 1 if nums[mid] < target
        return mid
    end

    -1
end

File: javascript/lesson 1/binarySearch.js
// https://leetcode.com/problems/binary-search

function search(nums, target) {
    let lo = 0;
    let hi = nums.length - 1;

    while (lo <= hi) {
        const mid = lo + ((hi - lo) >> 1);
        if (target < nums[mid]) {
            hi = mid - 1
        } else if (target > nums[mid]) {
            lo = mid + 1; 
        } else {
            return mid;
        }
    }

    return -1;
}
